packages/api/src/router/widgets/stocks.ts

```diff
@@ -0,0 +1,23 @@
+import { z } from "zod";
+
+import { fetchStockPriceHandler } from "@homarr/request-handler/stock-price";
+
+import { stockPriceTimeFrames } from "../../../../widgets/src/stocks";
+import { createTRPCRouter, publicProcedure } from "../../trpc";
+
+const stockPriceInputSchema = z.object({
+  stock: z.string().nonempty(),
+  timeRange: z.enum([...stockPriceTimeFrames.range]),
+  timeInterval: z.enum([...stockPriceTimeFrames.interval]),
+});
+
+export const stockPriceRouter = createTRPCRouter({
+  getPrices: publicProcedure.input(stockPriceInputSchema).query(async ({ input }) => {
+    const innerHandler = fetchStockPriceHandler.handler({
+      stock: input.stock,
+      timeRange: input.timeRange,
+      timeInterval: input.timeInterval,
+    });
+    return await innerHandler.getCachedOrUpdatedDataAsync({ forceUpdate: true });
```

Comment:
We should not force update here as this will result in a new call, even if there is a cached non stale value